File: cases/024.metric-recorder-buffered-flush/src/console_metric_recorder.cc
#include "console_metric_recorder.h"

namespace nitr::case024 {

ConsoleMetricRecorder::ConsoleMetricRecorder(std::ostream& out) : out_(out) {}

void ConsoleMetricRecorder::Record(const Metric& metric) {
  out_ << metric.name << '=' << metric.value << '\n';
}

void ConsoleMetricRecorder::Flush() { out_.flush(); }

}  // namespace nitr::case024

File: cases/032.metric-recorder-buffered-flush-python/src/console_metric_recorder.py
from __future__ import annotations

from typing import TextIO

from src.metric_recorder import Metric, MetricRecorder


class ConsoleMetricRecorder(MetricRecorder):
    def __init__(self, out: TextIO) -> None:
        self._out = out

    def record(self, metric: Metric) -> None:
        self._out.write(f"{metric.name}={metric.value}\n")
        self._out.flush()
